src/main/kotlin/com/openlattice/chronicle/study/StudyUpdate.kt

```diff
@@ -0,0 +1,32 @@
+package com.openlattice.chronicle.study
+
+import com.google.common.base.Preconditions
+import java.time.OffsetDateTime
+
+/**
+ * @author Solomon Tang <solomon@openlattice.com>
+ */
+data class StudyUpdate(
+    val title: String?,
+    val description: String?,
+    val startedAt: OffsetDateTime?,
+    val endedAt: OffsetDateTime?,
+    val lat: Double?,
+    val lon: Double?,
+    val group: String?,
+    val version: String?,
+    val settings: Map<String, Any>?,
+    val contact: String?,
+) {
+    init {
+        Preconditions.checkArgument(
+            title == null || title.isNotEmpty(),
+            "Title cannot be blank."
+        )
+        Preconditions.checkArgument(
+            contact == null || contact.isNotEmpty(),
+            "Contact cannot be blank."
+        )
```

Comment:
This doesn't work. checkArgument will throw illegal arg exception if the expression evaluates to false. If title/contact is null then the check will not fail, which I assume is the opposite of what we want

Looks like the correct boolean expression is `title != null && title.isNotBlank())`

Also, 
`.isNotBlank()` check will fail for strings consisting only of whitespace characters.
`.isEmpty()` check won't fail for such strings

Reply:
`title` and `contact` being null is ok because it means they have been omitted in the request, i.e.

```
{
  "group": "test",
  "version": "v1"
}
```

`isNotEmpty()` is also ok because we want the string to be not null, not empty `""`, and not whitespace only `"  "`.

Under the hood, `checkArgument` does `!expression` so the following is equivalent:

```
title == null || title.isNotEmpty()
!(title != null && !title.isNotEmpty()
```

Reply:
I was assuming that we want the string to be **both** null and contain characters other than white space. clarification makes sense. Only change necessary then is replacing `isNotEmpty` with `isNotBlank`